libyul/backends/evm/AbstractAssembly.h

```diff
@@ -99,7 +99,7 @@ class AbstractAssembly
 	/// Append the assembled size as a constant.
 	virtual void appendAssemblySize() = 0;
 	/// Creates a new sub-assembly, which can be referenced using dataSize and dataOffset.
-	virtual std::pair<std::shared_ptr<AbstractAssembly>, SubID> createSubAssembly(bool _creation, std::string _name = "") = 0;
+	virtual std::pair<std::shared_ptr<AbstractAssembly>, SubID> createSubAssembly(bool _creation, std::optional<uint8_t> _eofVersion, std::string _name = "") = 0;
```

Comment:
Also looking a bit closer here: should a sub-assembly get its own EOF version? I'd imagine that it'd always inherit the EOF version from the parent Assembly. (In the ``EthAssemblyAdapter`` implementation you can fetch it from the wrapped ``Assembly``, in ``NoOutputAssembly``, you'd need to store it.)

Or is the idea that technically there could be EOF version 1 containers containing EOF version 2 containers?